XR_APILAYER_NOVENDOR_toolkit/layer.cpp

```diff
@@ -1862,15 +1862,12 @@ namespace {
         }
 
         std::string getXrPath(XrPath path) {
-            std::string str;
-            if (path != XR_NULL_PATH) {
-                // TODO: I can't find in the spec if max path includes the trailing 0?
-                str.resize(XR_MAX_PATH_LENGTH + 1);
-                auto count = static_cast<uint32_t>(str.size());
-                CHECK_XRCMD(xrPathToString(GetXrInstance(), path, count, &count, &*str.begin())); // safe idiom
-                str.resize(size_t(count) - (count != 0));
-            }
-            return str;
+            uint32_t count;
+            CHECK_XRCMD(xrPathToString(GetXrInstance(), path, 0, &count, nullptr));
+            std::string s;
+            s.resize(count);
+            CHECK_XRCMD(xrPathToString(GetXrInstance(), path, count, &count, s.data()));
+            return s;
```

Comment:
I didn't investigate why the Varjo Runtime did fail when using a pre allocated static buffer of XR_MAX_PATH_LENGTH, but using the canonical way (get count then get data) should be bullet proof.

NB: we're allocating 1 extra trailing 0 here, but this has no other side effect in the code using the returned string (mostly == or find) and this protects from runtimes bugs (those that might be returning number of chars count without the trailing 0).